app/controllers/donors_controller.rb

```diff
@@ -8,14 +11,29 @@ def index
   end
 
   def admin_index
-    if current_donor.admin?
-      redirect_to '/'
+    @donor = Donor.where(approved: false, admin: false)
+    @donors = Donor.not_admin
+    @pickup = FoodPickup.find_by(id: params[:id])
+    @approved_pickups = FoodPickup.approved
+    @unapproved_pickups = FoodPickup.unapproved
+    @negotiable_profiles = Donor.where(negotiable: true)
+  end
+
+  def negotiate
+    @negotiable_profile = Donor.find_by(id: params[:id])
+  end
+
+  def update_rate
+    negotiable_profile = Donor.find_by(id: params[:id])
+    @rate = negotiable_profile.update(
+      charge: params[:charge],
+      negotiable: false
+    )
+    redirect_to '/admin'
```

Comment:
Why is the redirect up here? Shouldn't it happen once we know the update was successful?